### ISSUE TITLE
Huge Deferred Worker Queue 808

### ISSUE THREAD

**Issue by 4nanook** [friendica/friendica#10274]
- [x ] I have searched open and closed issues for duplicates

### Bug Description

I have 809 Items in the Deferred Worker Queue at present.  The machine this is running on is an i7-6850k 6-core 12-thread CPU clocked at 4.2 Ghz with 64GB of RAM dedicated to the web server.  The web server load averages about 3 which means at any given time on average only half the cores and a quarter of the threads are in use.  The database is Maria 10.5 running on an iofusion drive that supports atomic writes (recommended by MariaDB engineers for high usage).  The database is capable of handling 10,000 TPS, it is only averaging 400, so it is far from overload, slow queries are around 1-in-a-million.  I have Friendica configured for 200 parallel workers, maximum load average of 40 for worker, 250 for user.  When I inspect the worker queue it shows only items for the last minute, often completely empty.  When I inspect deferred worker queue it is huge.  I have workers scheduled out of cron once a minute.  It will fire off 40 of them but within 1-2 seconds all but 2-10 have exited.

free
              total        used        free      shared  buff/cache   available
Mem:       65862204    20017052    20253348      595432    25591804    44522468
Swap:     134217724           0   134217724

As you can see memory is not constrained, swap isn't even touched.

Actual Result:

809 Items presently in deferred worker queue.

Expected Result:

Very few items in deferred worker queue OR a lot of workers launched busily working on it.

### Screenshots
Not really a screen shot but a capture of queue:

<section><div id="adminpage">
	<h1>Administration - Inspect deferred worker queue (809)</h1>
	
	<p>This page lists the deferred worker jobs. These are jobs that couldn't initially be executed.</p>
	
ID | Command | Job parameters | Created | Priority
-- | -- | -- | -- | --
13238929 | APDelivery | wall-new: 2506601: https://friendica.wapnitsky.com/inbox: 16: {446146}: 1415781 | 2021-05-15 15:49:24 | 50
13239252 | APDelivery | wall-new: 2506728: https://friendica.wapnitsky.com/inbox: 16: {446146}: 1415847 | 2021-05-15 15:59:38 | 50
13239540 | APDelivery | wall-new: 2506850: https://friendica.wapnitsky.com/inbox: 16: {446146}: 1415906 | 2021-05-15 16:04:29 | 50
13239820 | APDelivery | wall-new: 2506966: https://friendica.wapnitsky.com/inbox: 1: {446146}: 1415957 | 2021-05-15 16:14:17 | 50
13239784 | APDelivery | wall-new: 2506966: https://toot.site/inbox: 1: {214250}: 1415957 | 2021-05-15 16:14:17 | 50
[...] | [...] | [...] | [...] | [...]

### Platform Info

Friendica Version:

This is Friendica, version 2021.06-dev that is running at the web location https://friendica.eskimo.com. The database version is 1414/1414, the post update version is 1400/1400.

Friendica Source: Github
PHP version: 7.4.18 (cli) (built: May  3 2021 11:27:06) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.18, Copyright (c), by Zend Technologies

SQL version: mysql Ver 15.1 Distrib 10.5.10-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**Comment by 4nanook** [comment]
Sorry there are three of these.  Github said page taking too long to load, hit reload, which I did.  Apparently it created the bug report but didn't finish displaying.

**Comment by AlfredSK** [comment]
My deferred queue is 11908. There is nothing to worry about. That's what that queue is about - queue the unsuccessful deliveries. They're unsuccessful because the remote side isn't reachable temporarily, is currently down, doesn't respond in the expected manner etc. The delivery will be tried again for some time with growing intervals until it is successful or until Friendica drops those delivery jobs.

**Comment by 4nanook** [comment]
Perhaps renaming this queue to something like deferred remote jobs so it is clear it is a remote and not a local issue?

**Comment by MrPetovan** [comment]
You can see that these jobs are deliveries, and the profile URLs are all remote, so I'm not sure adding the distinction in the name of the queue would provide any more information than there is already.

**Comment by 4nanook** [comment]
It wasn't clear to me what they were else I wouldn't have generated the ticket.  But since you've made it clear and apparently I'm the only one confused, I'll go ahead and close this out.

**Comment by 4nanook** [comment]
It wasn't clear to me what they were else I wouldn't have generated the ticket.  But since you've made it clear and apparently I'm the only one confused, I'll go ahead and close this out.

**Comment by MrPetovan** [comment]
I'm sorry I made you feel bad about this, my question wasn't rhetorical, I really wasn't sure what your suggestion would achieve. However, now that I've re-read your original issue post, I realize you were expecting these tasks to be handled at once. Deferred means that they are scheduled for later, and this information doesn't appear in this screen, and I believe this is the missing piece of information to understand what this list is about.

**Comment by 4nanook** [comment]
What was more of a question for me was the reason for deferral, is there
a shortage of resources on my end that I need to address?  I couldn't identify
any.  I did not understand that it is the remote end that it was waiting on.

-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
  Eskimo North Linux Friendly Internet Access, Shell Accounts, and Hosting.
    Knowledgeable human assistance, not telephone trees or script readers.
  See our web site: http://www.eskimo.com/ (206) 812-0051 or (800) 246-6874.

**Comment by MrPetovan** [comment]
I believe that with an explicit retry date in the future, you would have known why no worker was spawned to handle these tasks. It wouldn't have helped with this uncertainty about the job being remote or local but I believe there already is data to support this distinction, while there's no clear evidence these jobs will only be retried at a fixed later date and time.

**Comment by 4nanook** [comment]
Ok, well anything that makes it clear as to whether action is required
on my part or not is what I need.  It appears not.

-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
  Eskimo North Linux Friendly Internet Access, Shell Accounts, and Hosting.
    Knowledgeable human assistance, not telephone trees or script readers.
  See our web site: http://www.eskimo.com/ (206) 812-0051 or (800) 246-6874.

**Comment by tio-trom** [comment]
I keep on having over 60.000 of them...they look like this https://paste.trom.tf/boqelexodo.apache

My Friendica works really well, so if these are not an issue then all good. But it seems like a massive amount of them.
